### PR TITLE
integration test to cover yanked packages

### PR DESCRIPTION
**Pull request by slimreaper35**
# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Docs updated (if applicable)
- [ ] Docs links in the code are still valid (if docs were updated)

**Note:** if the contribution is external (not from an organization member), the CI
pipeline will not run automatically. After verifying that the CI is safe to run:

- [approve GitHub Actions workflows][approve-gh-actions] by clicking a button
- approve the Red Hat Trusted App Pipeline container build by commenting `/ok-to-test`
  (as is the [standard for Pipelines as Code][pac-running-pipeline])

[approve-gh-actions]: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
[pac-running-pipeline]: https://pipelinesascode.com/docs/guide/running/#running-the-pipelinerun

### CODE REVIEW

**Comment by slimreaper35** [comment]
Rebased on this [PR](https://github.com/containerbuildsystem/cachi2/pull/506)

I renamed cachito-pip-without-metadata and added this scenario so I do not create another repository.

File: tests/integration/test_pip.py [hunk 1]
```diff
@@ -73,8 +73,8 @@
         ),
         pytest.param(
             utils.TestParameters(
-                repo="https://github.com/cachito-testing/cachito-pip-without-metadata.git",
-                ref="4fb61a80f9bb19fdd422a74602b108d405ede299",
+                repo="https://github.com/cachito-testing/cachi2-pip-extra.git",
```
Comment (eskultety) [inline tests/integration/test_pip.py:76]: I know this is a GH organization to host test repositories, but you can't just rename publicly reachable repositories with no further notice, that's both confusing and a bad practice. What's worse is that you also force pushed the main branch overwriting history for everyone who may have the repository in question checked out locally - this is a big no no! We just lost history on the following commit: https://github.com/cachito-testing/cachi2-pip-extra/commit/e5e81c7 . I understand that all this was to address the fact that the original repository was tailored to a single specific use case, but in that case such a repository needs to be properly archived and a new one needs to be created to take its place to retain whole history.

**Review by eskultety — APPROVED** [review]
Since https://github.com/containerbuildsystem/cachi2/pull/506 is still open, this should have been made part of that PR, nevermind though this time, let's just make sure we merge both in a sequence.

**Comment by slimreaper35** [comment]
Should I move this to #506 ?

**Comment by eskultety** [comment]
> Should I move this to #506 ?

Nah, you've already got approvals there, you just need 1 more here, let's keep it as is.

**Comment by ejegrova** [comment]
@slimreaper35 please resolve conflicts

**Review by ejegrova — APPROVED** [review]

**Pull request merged**